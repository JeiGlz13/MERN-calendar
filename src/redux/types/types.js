export const types = {
    setLoginForm: "[LOGIN] set login form",
    setRegisterForm: "[LOGIN] set register form",

    uiOpenModal: "[UI] open modal",
    uiCloseModal: "[UI] close modal",

    eventAddNew: "[EVENT]  add new",
    eventSetActive: "[EVENT] set active",
    eventClearActiveEvent: "[EVENT] clear active",
    eventUpdate: "[EVENT] update",
    eventDeleted: "[EVENT] deleted",
    eventSetStartEnd: "[EVENT] set start-end",
    eventClearStartEnd: "[EVENT] clear start-end",
}