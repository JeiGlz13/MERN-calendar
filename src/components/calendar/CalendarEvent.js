import React from 'react';

export const CalendarEvent = ({event}) => {

    const {titulo, user} = event;
    return (
        <div className='flex flex-col focus:outline-none outline-none text-xs sm:text-sm md:text-base' >
            <span>{titulo}</span>
            <strong>- {user.name} </strong>
        </div>
    )
}
