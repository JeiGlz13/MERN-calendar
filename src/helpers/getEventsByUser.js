export const getEventsByUser = (eventos = [], uid) =>{
    return eventos.filter((event) => (event.user._id === uid) && event);
}