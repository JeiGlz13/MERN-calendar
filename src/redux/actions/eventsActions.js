import Swal from "sweetalert2";
import { fetchWithToken } from "../../helpers/fetch";
import { getEventsByUser } from "../../helpers/getEventsByUser";
import { prepareEvents } from "../../helpers/prepareEvents";
import { types } from "../types/types";

export const eventStartAddNew = (event) =>{
    return async (dispatch, getState) =>{
        const {uid, name} = getState().auth;

        try {
            const resp = await fetchWithToken('events', event, 'POST');
            const body = await resp.json();

            if (body.ok) {
                event.id = body.evento.id;
                event.user = {
                    _id: uid,
                    name: name
                }
                dispatch(eventAddNew(event))
            }
        } catch (error) {
            console.log(error);
        }
    }
}

const eventAddNew = (event) =>({
    type: types.eventAddNew,
    payload: event
});

export const eventSetActive = (event) =>({
    type: types.eventSetActive,
    payload: event
});

export const eventClearActive = () =>({
    type: types.eventClearActiveEvent
});

export const eventStartUpdate = (event) =>{
    return async (dispatch) =>{
        try {
            const resp = await fetchWithToken(`events/${event.id}`, event, 'PUT');
            const body = await resp.json();

            if (body.ok) {
                dispatch(eventUpdate(event));
            }else{
                Swal.fire('Error', body.msg ,'error');
            }
        } catch (error) {
            console.log(error);
        }
    }
} 

const eventUpdate= (event) =>({
    type: types.eventUpdate,
    payload: event
})

export const eventStartDelete = () =>{
    return async (dispatch, getState) =>{
        const {id} = getState().events.activeEvent;
        try {
            const resp = await fetchWithToken(`events/${id}`, {}, 'DELETE');
            const body = await resp.json();

            if (body.ok) {
                dispatch(eventDeleted());
                Swal.fire('Success', 'Evento eliminado con exito', 'success');
            }else{
                Swal.fire('Error', body.msg ,'error');
            }
        } catch (error) {
            console.log(error);
        }
    }
} 

const eventDeleted = () =>({type: types.eventDeleted});

export const eventSetStartEnd = (start, end) =>({
    type: types.eventSetStartEnd,
    payload: {
        eventStart: start,
        eventEnd: end
    }
});

export const eventClearEvents = () =>({type: types.eventClear});

export const eventClearStartEnd = () => ({type: types.eventClearStartEnd});

export const eventStartLoading = () =>{
    return async (dispatch, getState) =>{
        try {
                const {uid} = getState().auth;
               const resp = await fetchWithToken('events/');
               const body = await resp.json();

               const myEvents = getEventsByUser(body.eventos, uid);

               const eventos = prepareEvents(myEvents);


               dispatch(eventLoaded(eventos));


        } catch (error) {
            console.log(error)
        }
    }
}

const eventLoaded = (eventos) =>({
    type: types.eventLoaded,
    payload: eventos

})