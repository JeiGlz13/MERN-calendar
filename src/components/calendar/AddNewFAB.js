import React from 'react'
import { useDispatch } from 'react-redux'
import { openModal } from '../../redux/actions/modalActions';

export const AddNewFAB = () => {
    const dispatch = useDispatch();

    const setOpenModal = () =>{
        dispatch(openModal());
    }
    return (
        <button
        className= "absolute z-50 bottom-4 right-4 bg-indigo-500 text-white text-xl sm:text-2xl md:text-3xl h-10 w-10 sm:h-12 sm:w-12 md:h-16 md:w-16  rounded-full m-2"
        onClick={setOpenModal} >
            <i className='fas fa-plus'></i>
        </button>
    )
}
