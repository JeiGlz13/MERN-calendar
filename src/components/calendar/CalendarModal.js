import React, { useEffect, useState } from 'react';
import Modal from 'react-modal';

import DateTimePicker from 'react-datetime-picker';
import moment from 'moment';
import Swal from 'sweetalert2';
import { useDispatch, useSelector } from 'react-redux';
import { closeModal } from '../../redux/actions/modalActions'
import {eventAddNew, eventClearActive, eventClearStartEnd, eventUpdate} from '../../redux/actions/eventsActions';


Modal.setAppElement('#root');

const now = moment().minutes(0).seconds(0).add(1, "hours");
const nowPlus1 = now.clone().add(1, "hours");


export const CalendarModal = () => {
	const {modalOpen} = useSelector(state => state.modal);
	const {activeEvent, setStart, setEnd} = useSelector(state => state.events);

	let initEvent = {
		titulo: '',
		descripcion: '',
		start: now.toDate(),
		end: nowPlus1.toDate()
	}

	const dispatch = useDispatch();

	const [dateStart, setDateStart] = useState(now.toDate());
	const [dateEnd, setDateEnd] = useState(nowPlus1.toDate());

	const [formValues, setFormValues] = useState(initEvent);

	const {titulo, descripcion, start, end} = formValues;

	useEffect(() => {
		if(setStart && setEnd){
			initEvent = {
				...initEvent,
				start: setStart,
				end: setEnd
			}
			setDateStart(setStart);
			setDateEnd(setEnd);
		}else{
			const firstDate = now.toDate();
			const secondDate = nowPlus1.toDate();
			initEvent = {
				...initEvent,
				start: firstDate,
				end: secondDate
			}
				setDateStart(firstDate);
				setDateEnd(secondDate);
		}
	}, [setStart, setEnd])

	useEffect(() => {
		if (activeEvent) {
			setFormValues(activeEvent);
		}else{
			setFormValues(initEvent);
		}
		
	}, [activeEvent, setFormValues])


	const handleInputChange = ({target}) =>{
		setFormValues({
			...formValues,
			[target.name]: target.value
		})
	}

	
	const handleStartDate = (e) =>{
		setDateStart(e);
		setFormValues({
			...formValues,
			start: e
		});
	}

	const handleEndDate = (e) =>{
		setDateEnd(e);
		setFormValues({
			...formValues,
			end: e
		});
	}

	const validateForm = () =>{
		const firstDate = moment(start);
		const secondDate = moment(end);
		if(firstDate.isSameOrAfter(secondDate)){
			Swal.fire("Error", "La fecha de fin debe ser luego de la de inicio", "error");
			return false}
		if((titulo.trim().length < 2)||(descripcion.trim().length < 2)){
			Swal.fire("Error", "Rellene todos los campos", "error");
			return false}
		Swal.fire("Success", "Nueva nota añadida con éxito", "success");
		return true;
		
	}

	const saveNewEvent = (event) =>{
		dispatch( eventAddNew({
			...event,
			user: {
				id: new Date().getTime(),
				name: 'Jeisson'
			}
		}) );
	}

	const updateEvent = (evemt) =>{
		dispatch(eventUpdate(evemt));
	}

	const handleSubmitForm = (e) =>{
		e.preventDefault();
		if(validateForm() === true){
			//TODO: send to database

			if(activeEvent){
				updateEvent(formValues);
			}else{
				saveNewEvent(formValues);
			}
			setCloseModal();
		}
	}


	const setCloseModal = () =>{
       dispatch(closeModal());
	   setFormValues(initEvent);
	   dispatch(eventClearActive());
	   dispatch(eventClearStartEnd())
    }

    return (
        <Modal
        isOpen={modalOpen}
       // onAfterOpen={afterOpenModal}
        onRequestClose={setCloseModal}
        closeTimeoutMS={200}
        className="py-6 flex flex-col justify-center sm:py-12 outline-none focus:outline-none border-0"
        overlayClassName="modal-fondo"
      >
        
       
	<div className ="relative py-3 sm:max-w-xl sm:mx-auto font-Poppins">
		<div
			className ="absolute inset-0 bg-gradient-to-r from-indigo-300 to-indigo-600 shadow-lg transform -skew-y-3 sm:skew-y-0 -rotate-3 sm:-rotate-6 rounded-2xl sm:rounded-3xl">
		</div>
		<div className ="relative px-8 py-4 bg-white shadow-lg rounded-2xl sm:rounded-3xl sm:px-14 sm:py-8">
			<div className ="max-w-md mx-auto outline-none focus:outline-none border-0">
				<div>
					<h1 className ="text-lg sm:text-2xl font-semibold text-indigo-700 px-12 sm:px-24">
						{
							(activeEvent) ? ("Editar evento") : ("Nuevo Evento")
						}
					</h1>
				</div>
				<div className ="divide-y divide-gray-200">
					<div className ="py-4 sm:py-8 text-sm leading-6 space-y-4 text-gray-700 sm:text-lg sm:leading-7 flex flex-col justify-center">

                        <div className ="relative">
							<input autocomplete="off" id="titulo" name="titulo" type="text"
							value={titulo} onChange={handleInputChange}
							className ="peer placeholder-transparent h-10 w-full border-b-2 border-gray-300 text-gray-900 focus:outline-none focus:borer-rose-600" placeholder="Titulo" />
							<label for="titulo" className ="absolute left-0 -top-3.5 text-gray-800 text-sm peer-placeholder-shown:text-base peer-placeholder-shown:text-gray-440 peer-placeholder-shown:top-2 transition-all peer-focus:-top-3.5 peer-focus:text-indigo-600 peer-focus:text-sm">Titulo</label>
						</div>

						<div className ="relative">
							<label className ="absolute left-0 -top-3.5 text-gray-800 text-base">Fecha y hora inicio</label>

							<DateTimePicker
								onChange={handleStartDate}
								value={dateStart}
								className="mt-4"
							/>
						</div>

						<div className ="relative">
							<label className ="absolute left-0 -top-3.5 text-gray-800 text-base">Fecha y hora final</label>

							<DateTimePicker
								onChange={handleEndDate}
								value={dateEnd}
								minDate={dateStart}
								className="mt-4"
							/>
						</div>

                        <div className ="relative">
                            <textarea id="descripcion" name="descripcion" rows="5" 
							value={descripcion} onChange={handleInputChange}
							className ="resize-none peer placeholder-transparent w-full border-b-2 border-gray-300 text-gray-900 focus:outline-none focus:borer-rose-600" type = "text" placeholder="Descripcion">
                            </textarea>
							<label for="descripcion" className ="absolute left-0 -top-3.5 text-gray-800 text-sm peer-placeholder-shown:text-base peer-placeholder-shown:text-gray-440 peer-placeholder-shown:top-2 transition-all peer-focus:-top-3.5 peer-focus:text-indigo-600 peer-focus:text-sm">Descripcion</label>
						</div>
					
						<div className ="relative">
							<button className ="bg-indigo-500 text-white rounded-xl px-4 py-2"
							onClick={handleSubmitForm} >Guardar</button>
						</div>
					</div>
				</div>
			</div>
		</div>
	</div>
      </Modal>
    )
}
