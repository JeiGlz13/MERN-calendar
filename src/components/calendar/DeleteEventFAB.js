import React from 'react'
import { useDispatch } from 'react-redux'
import { eventStartDelete } from '../../redux/actions/eventsActions';

export const DeleteEventFAB = () => {
    const dispatch = useDispatch();

    const handleDelete = () =>{
        dispatch(eventStartDelete());
    }
    return (
        <button
        className='absolute z-50 bottom-4 left-4 bg-red-500 text-white text-xl sm:text-2xl md:text-3xl h-10 w-10 sm:h-12 sm:w-12 md:h-16 md:w-16 rounded-full m-2'
        onClick={handleDelete} >
            <i className='fas fa-trash'></i>
        </button>
    )
}
