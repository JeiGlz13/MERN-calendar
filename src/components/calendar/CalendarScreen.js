import React, { useState } from 'react'
import { Navbar } from '../ui/ui_navbar/Navbar';
import {Calendar, momentLocalizer} from 'react-big-calendar'
import moment from 'moment';
import 'moment/locale/es';

import 'react-big-calendar/lib/css/react-big-calendar.css';
import { messages } from '../../helpers/calendarMessagesEs';
import { CalendarEvent } from './CalendarEvent';
import { CalendarModal } from './CalendarModal';
import { useDispatch, useSelector } from 'react-redux';
import { openModal } from '../../redux/actions/modalActions';
import { eventClearActive, eventSetActive, eventSetStartEnd } from '../../redux/actions/eventsActions';
import { AddNewFAB } from './AddNewFAB';
import { DeleteEventFAB } from './DeleteEventFAB';

moment.locale('es');
const localizer = momentLocalizer(moment);

export const CalendarScreen = () => {
    const {events, activeEvent} = useSelector(state => state.events);
    const dispatch = useDispatch();
    const initialState = localStorage.getItem('lastView') || 'month';
    const [lastView, setLastView] = useState(initialState);

    const onDoubleClick = (e) =>{
        dispatch(openModal());
    }

    const onSelectEvent = (e) =>{
        dispatch(eventSetActive(e));
    }

    const onViewChange = (e) =>{
        setLastView(e);
        localStorage.setItem('lastView', e);
    }

    const eventStyleGetter = (event, start, end, isSelected) =>{
        const style = {
            backgroundColor: 'rgb(99 102 241)',
            borderRadius: '0px',
            opacity: 0.9,
            display: 'block',
            color: 'white'
        }

        return{
            style
        }
    }

    const onSelectSlot = (e) =>{
        dispatch(eventSetStartEnd(e.start, e.end))
        dispatch(eventClearActive());
        dispatch(openModal());
    }

    return (
        <div className='flex flex-col h-screen font-Poppins' >
            <Navbar/>

            <Calendar
                localizer={localizer}
                events={events}
                startAccessor="start"
                endAccessor="end"
                messages={messages}
                eventPropGetter={eventStyleGetter}
                onDoubleClickEvent={ onDoubleClick }
                onSelectEvent={onSelectEvent}
                onSelectSlot={onSelectSlot}
                selectable = {true}
                onView={onViewChange}
                view= {lastView}
                components={{
                    event: CalendarEvent
                }}
            />

            {
                (activeEvent)&&(
                    <DeleteEventFAB/>
                )
            }
            
            <AddNewFAB/>

            <CalendarModal/>
        </div>
    )
}
