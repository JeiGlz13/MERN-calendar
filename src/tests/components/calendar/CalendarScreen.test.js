import React from 'react';
import {mount} from 'enzyme';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';

import configureStore from 'redux-mock-store';
import thunk from "redux-thunk";
import {messages} from '../../../helpers/calendarMessagesEs';
import { CalendarScreen } from '../../../components/calendar/CalendarScreen';
import { types } from '../../../redux/types/types';
import { eventSetActive } from '../../../redux/actions/eventsActions';
import { act } from 'react-dom/cjs/react-dom-test-utils.production.min';

const middleware = [thunk];
const mockStore = configureStore(middleware);

jest.mock('../../../redux/actions/eventsActions', () => ({
    eventSetActive: jest.fn(),
    eventStartLoading: jest.fn()
}));

Storage.prototype.setItem = jest.fn();

const initState = {
    events: {
        events: []
    },
    auth: {
        uid: '123',
        name: 'Jeisson'
    },
    modal: {
        modalOpen: false
    }
};

const store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
    <Provider store={store}>
        <CalendarScreen/>
    </Provider>
);

describe('Pruebas en el calendar screen', () => {
    test('Debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    test('Pruebas con las interacciones del calendario', () => {
        const calendar = wrapper.find('Calendar');
        const calendarMessages = calendar.prop('messages');
        expect(calendarMessages).toEqual(messages);

        calendar.prop('onDoubleClickEvent')();
        expect(store.dispatch).toHaveBeenCalledWith({
            type: types.uiOpenModal
        });

        calendar.prop('onSelectEvent')({
            start: 'Hola'
        });

        expect(eventSetActive).toHaveBeenCalledWith({
            start: 'Hola'
        });

        act(()=>{
            calendar.prop('onView')('week');
            expect(localStorage.setItem).toHaveBeenCalledWith('lastView', 'week');
        });

    });
    
});
