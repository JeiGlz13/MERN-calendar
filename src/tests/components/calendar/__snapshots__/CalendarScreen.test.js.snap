// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el calendar screen Debe mostrarse correctamente 1`] = `
<div
  className="flex flex-col h-screen font-Poppins"
>
  <div
    className="flex flex-wrap box-border"
  >
    <section
      className="relative mb-4 w-full"
    >
      <nav
        className="flex justify-between bg-zinc-800 text-white w-full"
      >
        <div
          className="px-5 xl:px-12 py-3 flex w-full items-center justify-between"
        >
          <h3
            className="text-lg sm:text-xl md:text-2xl font-bold font-heading"
          >
            Hola, 
            Jeisson
          </h3>
          <div
            className="flex items-center space-x-5"
          >
            <button
              className="px-5 py-2 rounded-xl text-base sm:text-lg md:text-xl font-medium text-white bg-indigo-500 hover:bg-indigo-600 active:bg-grey-900 focus:outline-none transition-all
 flex items-center"
              onClick={[Function]}
            >
              <i
                className="fas fa-door-open mr-2 text-base sm:text-lg md:text-xl"
              />
              Salir
            </button>
          </div>
        </div>
      </nav>
    </section>
  </div>
  <div
    className="rbc-calendar"
  >
    <div
      className="rbc-toolbar"
    >
      <span
        className="rbc-btn-group"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          Hoy
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          &lt;
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          &gt;
        </button>
      </span>
      <span
        className="rbc-toolbar-label"
      >
        enero 2022
      </span>
      <span
        className="rbc-btn-group"
      >
        <button
          className="rbc-active"
          key="month"
          onClick={[Function]}
          type="button"
        >
          Mes
        </button>
        <button
          className=""
          key="week"
          onClick={[Function]}
          type="button"
        >
          Semana
        </button>
        <button
          className=""
          key="day"
          onClick={[Function]}
          type="button"
        >
          Día
        </button>
        <button
          className=""
          key="agenda"
          onClick={[Function]}
          type="button"
        >
          Agenda
        </button>
      </span>
    </div>
    <div
      aria-label="Month View"
      className="rbc-month-view"
      role="table"
    >
      <div
        className="rbc-row rbc-month-header"
        role="row"
      >
        <div
          className="rbc-header"
          key="header_0"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            lun.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_1"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            mar.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_2"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            mié.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_3"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            jue.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_4"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            vie.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_5"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            sáb.
          </span>
        </div>
        <div
          className="rbc-header"
          key="header_6"
        >
          <span
            aria-sort="none"
            role="columnheader"
          >
            dom.
          </span>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_0"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                27
              </a>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_1"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                28
              </a>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_2"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                29
              </a>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_3"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                30
              </a>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_4"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                31
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_5"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                01
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_6"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                02
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg rbc-today"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell rbc-now rbc-current"
              key="header_0"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                03
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                04
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                05
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_3"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                06
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_4"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                07
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_5"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                08
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_6"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                09
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell"
              key="header_0"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                10
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                11
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                12
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_3"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                13
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_4"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                14
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_5"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                15
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_6"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                16
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell"
              key="header_0"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                17
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                18
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                19
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_3"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                20
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_4"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                21
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_5"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                22
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_6"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                23
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell"
              key="header_0"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                24
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_1"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                25
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_2"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                26
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_3"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                27
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_4"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                28
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_5"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                29
              </a>
            </div>
            <div
              className="rbc-date-cell"
              key="header_6"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                30
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="rbc-month-row"
        role="rowgroup"
      >
        <div
          className="rbc-row-bg"
        >
          <div
            className="rbc-day-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
          <div
            className="rbc-day-bg rbc-off-range-bg"
          />
        </div>
        <div
          className="rbc-row-content"
          role="row"
        >
          <div
            className="rbc-row "
          >
            <div
              className="rbc-date-cell"
              key="header_0"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                31
              </a>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_1"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                01
              </a>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_2"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                02
              </a>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_3"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                03
              </a>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_4"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                04
              </a>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_5"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                05
              </a>
            </div>
            <div
              className="rbc-date-cell rbc-off-range"
              key="header_6"
              role="cell"
            >
              <a
                href="#"
                onClick={[Function]}
                role="cell"
              >
                06
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <button
    className="absolute z-50 bottom-4 right-4 bg-indigo-500 text-white text-xl sm:text-2xl md:text-3xl h-10 w-10 sm:h-12 sm:w-12 md:h-16 md:w-16  rounded-full m-2"
    onClick={[Function]}
  >
    <i
      className="fas fa-plus"
    />
  </button>
  <Portal
    containerInfo={
      <div
        class="ReactModalPortal"
      />
    }
  />
</div>
`;
